src/Action/SetObjectFieldValueAction.php

```diff
@@ -115,7 +115,20 @@ public function __invoke(Request $request): JsonResponse
 
         // Handle date and datetime types have setter expecting a DateTime object
         if ('' !== $value && \in_array($fieldDescription->getType(), ['date', 'datetime'], true)) {
-            $value = new \DateTime($value);
+            $inputTimezone = new \DateTimeZone(date_default_timezone_get());
+            $outputTimezone = $fieldDescription->getOption('timezone');
+
+            if (\is_string($outputTimezone)) {
```

Comment:
IMO, the validation about allowed types should be done at field description.

Reply:
Allowed values for timezone option is `\DateTimeZone|string|false|null`
https://github.com/twigphp/Twig/blob/3.x/src/Extension/CoreExtension.php#L410